core/src/replay_stage.rs

```diff
@@ -81,6 +81,16 @@ pub enum HeaviestForkFailures {
     NoPropagatedConfirmation(u64),
 }
 
+pub enum VoteOp {
+    PushVote(Vec<Slot>),
+    RefreshVote(Slot),
+}
+
+pub struct VoteInfo {
+    pub vote_tx: Transaction,
+    pub op: VoteOp,
+}
+
```

Comment:
I think a single enum would be simpler:
```
pub enum VoteOp {
    Push { vote: Transaction, tower: Vec<Slot> },
    Refresh { vote: Transaction, slot: Slot },
}
```
it will also be more explicit what `Vec<Slot>` for example is.

also, shouldn't these type definitions go in `core/src/voting_service.rs`?